src/api/types/types.ts

```diff
@@ -1,100 +1,134 @@
+  timespent: string|null;
+  subtasks: [];
+  aggregateprogress: Progress;
+  timeoriginalestimate: null;
+  security: null;
+  resolution: null;
+  lastViewed: null;
+  timeestimate: null;
+  aggregatetimespent: null;
+  aggregatetimeoriginalestimate: null;
+  aggregatetimeestimate: null;
+  environment: null;
+  progress: Progress;
+} & CustomFieldsValues;
+
+export type IssueBean = RequiredProps<
+  components["schemas"]["IssueBean"],
+  "id"|"key"|"self"|"expand"
+> & {
+  fields: FieldsValues;
+  editmeta: {
+    fields: Record<FieldMeta["key"], FieldMeta>;
   };
-  // ToDo: need typings
-  dataDependencies?: any;
-  hasGeneratedIssueFormSuccessfully?: boolean;
-  isUnlinkingIssue?: boolean;
-  issueComments?: Record<string, JiraComment[]>;
-  _error?: Error | unknown;
-}
+};
 
-export type Action =
-  | { type: "changePage"; page: Page; params?: object }
-  | { type: "loadContext"; context: Context }
-  | { type: "linkIssueSearchListLoading" }
-  | { type: "linkIssueSearchList"; list: IssueSearchItem[] }
-  | { type: "linkIssueSearchListReset" }
-  | { type: "linkedIssuesListLoading" }
-  | { type: "linkedIssuesList"; list: IssueItem[] }
-  | { type: "unlinkIssue"; key: string }
-  | { type: "issueAttachmentsLoading" }
-  | { type: "issueAttachments"; key: string; attachments: IssueAttachment[] }
-  | { type: "loadDataDependencies"; deps: any }
-  | { type: "failedToGenerateIssueForm" }
-  | { type: "error"; error: string }
-  | { type: "issueComments"; key: string; comments: JiraComment[] };
-
-export interface TicketContext extends Context {
-  data: { ticket: { id: string; permalinkUrl: string; subject: string } };
-}
+export type SearchIssues = Omit<components["schemas"]["SearchResults"], "issues"> & {
+  issues: IssueBean[];
+};
 
-export interface SearchParams {
-  withSubtask?: boolean;
-  projectId?: string;
-}
+export type AvatarUrls = Required<components["schemas"]["AvatarUrlsBean"]>;
 
-export interface IssueItem {
-  id: number;
+export type SearchIssueItem = Omit<
+  FieldsValues,
+  "summary"|"status"
+> & {
+  id: string;
   key: string;
+  keyHtml?: string;
   summary: string;
-  projectKey: string;
-  projectName: string;
+  summaryHtml?: string;
   status: string;
-  reporterId: string;
-  reporterName: string;
-  reporterAvatarUrl: string;
-  assigneeId: string;
-  assigneeName: string;
-  assigneeAvatarUrl: string;
+  projectKey?: string;
+  projectName?: string;
+  reporterId?: string;
+  reporterName?: string;
+  reporterAvatarUrl?: string;
   epicKey?: string;
   epicName?: string;
-  priority: string;
-  priorityId: string;
-  priorityIconUrl: string;
-  sprints?: {
-    sprintBoardId?: number;
-    sprintName?: string;
-    sprintState?: string;
-  }[];
+  linkedCount?: number;
+};
+
+export type IssueItem = SearchIssueItem & {
+  assigneeId?: string;
+  assigneeName?: string;
+  assigneeAvatarUrl?: string;
+  priority?: Priority;
+  priorityId?: string;
+  priorityIconUrl?: string;
+  sprints?: Array<{
```

Comment:
Why is the `Array<...>` style used here but `...[]` is used elsewhere.

Reply:
In this case, it's simply easier for reading and understanding the code. Fixed to `...[]`